### PR TITLE
Fixes Ragin' Mages... m-maybe?

### PR DESCRIPTION
**Pull request by FreeStylaLT**
Since.. it's entirely dependent on playercount, I couldn't _actually_ properly test it. BUT in local testing, in theory, it worked as it should (1 player in the round ended up with 1 wizard whose death ended the round).

So I am just kind of posting this here to get looked at by people who can read code better and see if it would work with higher amounts.

### CODE REVIEW

**Comment by Fox-McCloud** [comment]
@Crazylemon64

File: code/game/gamemodes/wizard/raginmages.dm [hunk 1]
```diff
@@ -179,7 +181,7 @@
 	return new_character
 
 /datum/game_mode/wizard/raginmages/declare_completion()
-	if(finished)
+	if(!wizard_cap)
 		feedback_set_details("round_end_result","loss - wizard killed")
 		to_chat(world, "\red <FONT size = 3><B> The crew has managed to hold off the wizard attack! The Space Wizards Federation has been taught a lesson they will not soon forget!</B></FONT>")
```
Comment (FreeStylaLT) [inline code/game/gamemodes/wizard/raginmages.dm]: hisss \red

File: code/game/gamemodes/wizard/raginmages.dm [hunk 2]
```diff
@@ -25,14 +26,13 @@
 	var/obj_count = 1
 	for(var/datum/objective/objective in wizard.objectives)
 		to_chat(wizard.current, "<B>Objective #[obj_count]</B>: [objective.explanation_text]")
-		obj_count++
-	to_chat(wizard.current, "<b>Objective Alpha</b>: Make sure the station pays for its actions against our diplomats")
 	return
```
Comment (Crazylemon64) [inline code/game/gamemodes/wizard/raginmages.dm]: Why are you removing this line?

File: code/game/gamemodes/wizard/raginmages.dm [hunk 3]
```diff
@@ -121,23 +121,25 @@
...
-		if(!candidates.len)
-			message_admins("No applicable clients for the next ragin' mage, asking ghosts instead.")
-			var/time_passed = world.time
-			for(var/mob/dead/observer/G in player_list)
-				if(!jobban_isbanned(G, "wizard") && !jobban_isbanned(G, "Syndicate"))
-					spawn(0)
-						switch(alert(G, "Do you wish to be considered for the position of Space Wizard Foundation 'diplomat'?","Please answer in 30 seconds!","Yes","No"))
-							if("Yes")
-								if((world.time-time_passed)>300)//If more than 30 game seconds passed.
+	if(wizard_cap)
```
Comment (Crazylemon64) [inline code/game/gamemodes/wizard/raginmages.dm]: `wizard_cap` will always be 0 with your current code

File: code/game/gamemodes/wizard/raginmages.dm [hunk 4]
```diff
@@ -25,14 +26,13 @@
...
-	to_chat(wizard.current, "<b>Objective Alpha</b>: Make sure the station pays for its actions against our diplomats")
 	return
 
 /datum/game_mode/wizard/raginmages/check_finished()
 	var/wizards_alive = 0
 	// Accidental pun!
-	var/wizard_cap = (max_mages || (num_players_started() / players_per_mage))
+	var/max_mages = (num_players_started() / players_per_mage)
+	var/wizard_cap = max_mages
 	for(var/datum/mind/wizard in wizards)
```
Comment (Crazylemon64) [inline code/game/gamemodes/wizard/raginmages.dm]: These variables are being declared locally - the ones you have on the datum are not being referenced within here

**Comment by Crazylemon64** [comment]
What exactly is the problem with Ragin' mages you are attempting to address?

**Comment by FreeStylaLT** [comment]
The fact that it only ends when the shuttle docks, rather than when the wizards run out. Right now they spawn infinitely until round end, from what I can tell.

**Comment by Earthdivine** [comment]
That's what Ragin Mages kinda are meant to be. An onslaught of attacks with room to breathe in between them. If they crew doesn't want to keep up with it they can always call the shuttle.

File: code/game/gamemodes/wizard/raginmages.dm [hunk 5]
```diff
@@ -179,7 +181,7 @@
 	return new_character
 
 /datum/game_mode/wizard/raginmages/declare_completion()
-	if(finished)
+	if(!wizard_cap)
```
Comment (Crazylemon64) [inline code/game/gamemodes/wizard/raginmages.dm]: Your code as-is will instantly finish

**Comment by FreeStylaLT** [comment]
That's nnnot how it's meant to be used, at least judging by the fact max_mages, wizard_cap and players_per_mage variables _exist_ at all and by how they're used

**Comment by FreeStylaLT** [comment]
Erhrh, screw it, I am not entirely sure I am going about fixing this the right way, neither am I sure if it would even _work_ with higher playercounts, nor am I skilled enough to do this in the first place! Bad idea on all accounts.

**Comment by Crazylemon64** [comment]
@FreeStylaLT How many mages are being spawned? It SHOULD be 1/6 of the people in the game - so a round with 60 people would have 10 wizards, a wizard spawning every 7 minutes after half an hour - or if all the other wizards are dead. So, if the first wizard survives, there would be a wizard spawn, 30 minutes in, persisting for the next 7*10 minutes - so wizards would spawn until 1:40 (100 minutes) in.

**Pull request closed without merging**